### PR TITLE
UCP/KEEPALIVE: skip progress iterations

### PR DESCRIPTION
**Pull request by hoopoepg**
- added parameter to skip progress iterations to reduce ucs_get_time call

### CODE REVIEW

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -32,6 +32,8 @@
 #include <sys/epoll.h>
 
 
+#define UCP_WORKER_KEEPALIVE_ITER_SKIP 32
```
Comment (hoopoepg) [inline src/ucp/core/ucp_worker.c:35]: @yosefe make this parameter run-time configurable?

Comment (yosefe) [reply src/ucp/core/ucp_worker.c:35]: IMO, the benefit of making it fast check (mod by constant) outweighs the value of making it configurable: we already have a configurable timeout, this is just to reduce the cost of timer check

File: src/ucp/core/ucp_worker.c [hunk 2]
```diff
@@ -2688,12 +2691,19 @@ ucp_worker_keepalive_next_ep(ucp_worker_h worker)
...
     ucp_worker_h worker = (ucp_worker_h)arg;
-    ucs_time_t now      = ucs_get_time();
+    ucs_time_t now;
     ucs_list_link_t *iter_begin;
     ucp_ep_h ep;
 
     ucs_assert(worker->context->config.ext.keepalive_num_eps != 0);
 
+    worker->keepalive.iter_skip++;
+    if (worker->keepalive.iter_skip < UCP_WORKER_KEEPALIVE_ITER_SKIP) {
```
Comment (yosefe) [inline src/ucp/core/ucp_worker.c:2701]: if ((worker->keepalive.iter_count++ % UCP_WORKER_KEEPALIVE_ITER_SKIP) != 0)

**Comment by yosefe** [comment]
@leibin2014 can you pls take a look?

File: src/ucp/core/ucp_worker.c (hunk 2 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_worker.c]: updated

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Comment by brminich** [comment]
does value 32 improve iodemo performance? Was any other value checked?

**Comment by hoopoepg** [comment]
> does value 32 improve iodemo performance? Was any other value checked?

yes, on my tests keepalive_progress consumption is reduced to less than 1%

**Comment by hoopoepg** [comment]
@yosefe backport it to v1.10?

**Comment by yosefe** [comment]
Is KA enabled by default in v1.10?

**Comment by hoopoepg** [comment]
no

**Comment by yosefe** [comment]
so no need

**Pull request merged**